### PR TITLE
Aut 20 install sumologic docker

### PR DESCRIPTION
**Pull request by shinitiandrei**
Sumologic installation in docker containers.
Supports:
- Redhat
- Amazon Linux
- Suse

Unit and integrations tests working.

Story:
https://soltiusnz.atlassian.net/browse/AUT-20

### CODE REVIEW

File: recipes/linux.rb [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+RHEL_COLLECTOR =            'sumo_rhel_logfiles'.freeze
+SUSE_COLLECTOR =            'sumo_sles_logfiles'.freeze
+SUSE_JOURNALCTL_COLLECTOR = 'sumo_sles_journalctl'.freeze
+AMAZON_COLLECTOR =          'sumo_amazon_logfiles'.freeze
+
+#
+# Instal/Configure Docker
+#
+docker_installation 'docker' do
```
Comment (igorlg) [inline recipes/linux.rb]: Move the Docker installation to a separate recipe called `docker`. Include this recipe here.

File: recipes/linux.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  tag 'latest-file'
+  action :pull
+end
+
+## TODO: Refactor the code below is too long and sometimes redundant.
+## Ideally we need to create a "main" body code then alter what needs to
+## be changed regarding each OS
+case node['platform']
+when 'redhat'
+  docker_container 'sumologic-collector-file' do
```
Comment (igorlg) [inline recipes/linux.rb]: Too much repeated code... See what's common and isolate it...

File: recipes/linux.rb [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      "SUMO_ACCESS_ID=#{node['sumologic']['access_id']}",
+      "SUMO_ACCESS_KEY=#{node['sumologic']['access_key']}",
+      "SUMO_COLLECTOR_NAME=#{AMAZON_COLLECTOR}"
+    ]
+    restart_policy 'always'
+    action :run
+  end
+
+when 'suse'
+  docker_image 'sumologic/collector' do
```
Comment (igorlg) [inline recipes/linux.rb]: why `docker_image` is only created here? Isn't it required for all other OSs as well?

File: recipes/linux.rb [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    env [
+      "SUMO_ACCESS_ID=#{node['sumologic']['access_id']}",
+      "SUMO_ACCESS_KEY=#{node['sumologic']['access_key']}",
+      "SUMO_COLLECTOR_NAME=#{SUSE_JOURNALCTL_COLLECTOR}"
+    ]
+    restart_policy 'always'
+    action :run
+  end
+
+  file '/etc/systemd/system/journalctl_sumo.service' do
```
Comment (igorlg) [inline recipes/linux.rb]: Store the file contents as a _file_ (inside _files_/) or as a ERB _template_  (inside _templates/_)

File: recipes/linux.rb [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    RestartSec=5s
+
+    [Install]
+    WantedBy=multi-user.target
+    "
+    mode 0o755
+    action :create
+  end
+
+  execute 'Created Sumologic Sysv Service File' do
```
Comment (igorlg) [inline recipes/linux.rb]: Use Chef resource `service` for enabling these services.

File: test/unit/recipes/linux_spec.rb [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+                  'SUMO_ACCESS_ID=abcdefghijklmnopqrstuvwxyz',
+                  'SUMO_ACCESS_KEY=q1ew321ew2154w5412r5y7ey21',
+                  'SUMO_COLLECTOR_NAME=sumo_sles_journalctl'
+                ]
+              ]
+            end
+            it 'creates file journalctl_sumo.service in /etc/systemd/system/' do
+              expect(chef_run).to create_file('/etc/systemd/system/journalctl_sumo.service').with(mode: 0o755)
+              expect(chef_run).to create_file('/etc/systemd/system/journalctl_sumo.service').with(content: "
+    [Unit]
```
Comment (igorlg) [inline test/unit/recipes/linux_spec.rb]: Store the file contents here using

```let(:file_journalctl_sumo_service) { <string> }```

File: recipes/linux.rb (hunk 2 above)
Comment (shinitiandrei) [inline recipes/linux.rb]: Yeah I know, that's what I have commented there, I will change it then

File: recipes/linux.rb (hunk 3 above)
Comment (shinitiandrei) [inline recipes/linux.rb]: This specific image is just for suse, as it requires another image, if you see above the code there is a "tag: latest-file" and this one is common for everyone.
And the suse uses latest-file and latest-syslog, so this block it is just for the latest-syslog.

**Comment by shinitiandrei** [comment]
Suse is being removed for the meanwhile as we need to check why is that failing on travis, but it does not fail when spinning up a new instance locally.
All recipes/integration tests are working and tested.

**Comment by igorlg** [comment]
@shinitiandrei pls open a bug on JIRA to investigate this...

**Comment by shinitiandrei** [comment]
Added bug issue AUT-45

**Pull request merged**